prov/efa/src/rxr/rxr_ep.c

```diff
@@ -2154,9 +2154,12 @@ void rxr_ep_progress_internal(struct rxr_ep *ep)
 	dlist_foreach_container_safe(&ep->read_pending_list, struct rxr_read_entry,
 				     read_entry, pending_entry, tmp) {
 		peer = rxr_ep_get_peer(ep, read_entry->addr);
-		assert(peer);
```

Comment:
What change beyond this patch can cause `peer` can be NULL? I mean If it could be NULL, why this assert does not fail before?

Reply:
This happens when a local read request was queued. I encountered this issue when running fabtest from host sending buffer to GPU receiving buffer.

Reply:
So, it's just we did not test this path before, right?

Reply:
Yes

Reply:
I see, thank you!